src/incidentFeed.test.ts

```diff
@@ -0,0 +1,81 @@
+const firstResponse = {
+  items: [
+    { link: 'id=1', content: 'not updated' },
+    { link: 'id=2', content: 'not updated' }
+  ]
+};
+const secondResponse = {
+  items: [
+    { link: 'id=1', content: 'not updated' },
+    { link: 'id=2', content: 'not updated' },
+    { link: 'id=3', content: 'not updated' }
+  ]
+};
+const thirdResponse = {
+  items: [
+    { link: 'id=1', content: 'updated' },
+    { link: 'id=2', content: 'not updated' },
+    { link: 'id=3', content: 'not updated' }
+  ]
+};
+const fourthResponse = {
+  items: [
+    { link: 'id=1', content: 'updated' },
+    { link: 'id=2', content: 'not updated' },
+    { link: 'id=3', content: 'not updated' }
+  ]
+};
+
+jest.doMock(
+  'rss-parser',
+  () =>
+    // has to use function keyword to be called with new keyword (ie act as a constructor)
+    /* tslint:disable-line only-arrow-functions */ function() {
+      return {
+        parseURL: jest
+          .fn()
+          .mockReturnValueOnce(firstResponse)
+          .mockReturnValueOnce(secondResponse)
+          .mockReturnValueOnce(thirdResponse)
+          .mockReturnValueOnce(fourthResponse)
+      };
+    }
+);
+
+import IncidentFeed from './incidentFeed';
+const incidentFeed = new IncidentFeed();
+
+describe('IncidentFeed', () => {
+  test(':getFeed returns current feed without fetching and parsing an update', () => {
+    expect(incidentFeed.getFeed()).toMatchObject([]);
+    expect(incidentFeed.getFeed()).not.toMatchObject(['somee value']);
+  });
+  test('first update', async () => {
+    await incidentFeed.update();
+    expect(incidentFeed.getFeed()).toBe(firstResponse.items);
+    expect(incidentFeed.newIncidents()).toMatchObject([]);
+    expect(incidentFeed.updatedIncidents()).toMatchObject([]);
+  });
+  test('second update', async () => {
+    await incidentFeed.update();
+    expect(incidentFeed.getFeed()).toBe(secondResponse.items);
+    expect(incidentFeed.newIncidents()).toMatchObject([
+      { link: 'id=3', content: 'not updated' }
+    ]);
+    expect(incidentFeed.updatedIncidents()).toMatchObject([]);
+  });
+  test('third update', async () => {
+    await incidentFeed.update();
+    expect(incidentFeed.getFeed()).toBe(thirdResponse.items);
+    expect(incidentFeed.newIncidents()).toMatchObject([]);
+    expect(incidentFeed.updatedIncidents()).toMatchObject([
+      { link: 'id=1', content: 'updated' }
+    ]);
+  });
+  test('fourth update', async () => {
+    await incidentFeed.update();
+    expect(incidentFeed.getFeed()).toBe(fourthResponse.items);
```

Comment:
You were right - stripping back incidentFeed made it much easier to write a test 👍